std/os.zig

```diff
@@ -756,6 +756,44 @@ pub fn getEnvMap(allocator: *Allocator) !BufMap {
 
             try result.setMove(key, value);
         }
+    } else if (builtin.os == Os.wasi) {
+        var environ_count: usize = undefined;
+        var environ_buf_size: usize = undefined;
+
+        const environ_sizes_get_ret = std.os.wasi.environ_sizes_get(&environ_count, &environ_buf_size);
+        if (environ_sizes_get_ret != os.wasi.ESUCCESS) {
+            return unexpectedErrorPosix(environ_sizes_get_ret);
+        }
+
+        // TODO: Verify that the documentation is incorrect
+        // https://github.com/WebAssembly/WASI/issues/27
+        var environ = try allocator.alloc(?[*]u8, environ_count + 1);
+        defer allocator.free(environ);
+        var environ_buf = try std.heap.wasm_allocator.alloc(u8, environ_buf_size);
+        defer allocator.free(environ_buf);
+
+        const environ_get_ret = std.os.wasi.environ_get(environ.ptr, environ_buf.ptr);
+        if (environ_get_ret != os.wasi.ESUCCESS) {
+            return unexpectedErrorPosix(environ_get_ret);
+        }
+
+        for (environ) |env| {
+            if (env) |ptr| {
+                var line_i: usize = 0;
+                while (ptr[line_i] != 0 and ptr[line_i] != '=') : (line_i += 1) {}
```

Comment:
I'm fine with this, but you might consider using `std.mem.separate`

Reply:
Combining this and the other comment about `assert`, you could do this:

```zig
const line = std.mem.toSlice(u8, ptr);
var it = std.mem.separate(line, "=");
const key = it.next().?;
const value = it.next().?;
```

Reply:
Done. This also cleanly handles wasmer.